### PR TITLE
[harmonyhub] Fix reliability issues

### LINKED ISSUE #13701: [harmonyhub] Reliability issues
After upgrading router firmware, causing temporary network disruptions, the Harmony Binding did not recover well.

Secondly, from time to time I experience that the binding needs to be restarted. I don't have hard evidence for this case, but it might be related to the first one. First noticed: https://community.openhab.org/t/openhab-2-5-harmony-hub-binding-stability-issues/, although I believe stability was improved since then.

## Expected Behavior

- The binding should reconnect after being disconnected.
- Running jobs should be interrupted when binding is reconfigured/restarted.

## Current Behavior
After network disruption, this problem was recurring:
```
2022-11-11 23:51:29.461 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HarmonyHubHandler of thing harmonyhub:hub:home tried updating channel currentActivity although the handler was already disposed.
2022-11-11 23:51:29.462 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HarmonyHubHandler tried updating thing harmonyhub:hub:home although the handler was already disposed.
```

## Possible Solution
After looking into the code, I found some issues. I will provide a PR with proposed fixes.

## Steps to Reproduce
Turn off router for a few minutes and turn it on again. I have no confirmation yet if this is guaranteed to reproduce the problem.

### PR DESCRIPTION
**Pull request by jlaur**
Publishing as draft until I have run some additional tests and observed reliability for at least some days:

- [x] Restart router.
- [x] Restart hub.
- [x] Reconfigure thing.
- [x] Restart binding.

Main issue fixed:
```
2022-11-11 23:51:29.461 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HarmonyHubHandler of thing harmonyhub:hub:home tried updating channel currentActivity although the handler was already disposed.
2022-11-11 23:51:29.462 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler HarmonyHubHandler tried updating thing harmonyhub:hub:home although the handler was already disposed.
```

Fixes #13701

### CODE REVIEW

**Comment by digitaldan** [comment]
Thanks @jlaur !  I'll try running it as well this weekend.

**Comment by jlaur** [comment]
Router reboot test passed, but there might be room for further improvement:
```
2022-11-12 22:23:48.481 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'harmonyhub:hub:home' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): Hearbeat failed
2022-11-12 22:23:48.485 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'harmonyhub:device:home:lgtv' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2022-11-12 22:24:48.519 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'harmonyhub:hub:home' changed from OFFLINE (COMMUNICATION_ERROR): Hearbeat failed to OFFLINE (COMMUNICATION_ERROR): Could not connect: Can not call connect on already connected session
2022-11-12 22:28:48.649 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'harmonyhub:hub:home' changed from OFFLINE (COMMUNICATION_ERROR): Could not connect: Can not call connect on already connected session to ONLINE
2022-11-12 22:28:48.652 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'harmonyhub:device:home:lgtv' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
```

I'll have a look.

**Comment by jlaur** [comment]
@digitaldan - I went through all the tests listed and have run it since PR was created without any issues. If you are interested, you might want to have a look at my comment above. This happens outside of the binding (in your library), but I don't know if it's worth doing anything about.

File: bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/discovery/HarmonyHubDiscoveryService.java [hunk 1]
```diff
@@ -171,6 +171,7 @@ private void sendDiscoveryMessage(String discoverString) {
             // Broadcast the message over all the network interfaces
             Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
             while (interfaces.hasMoreElements()) {
+                @Nullable
```
Comment (lolodomo) [inline bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/discovery/HarmonyHubDiscoveryService.java:174]: Is it necessary? 
You add nullable but then you do not check if the value is null.

Comment (jlaur) [reply bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/discovery/HarmonyHubDiscoveryService.java:174]: See https://github.com/openhab/openhab-addons/pull/12303#discussion_r810625052

Comment (lolodomo) [reply bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/discovery/HarmonyHubDiscoveryService.java:174]: Ok, strictly the same thing.

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**